modules/openapi-generator/src/main/resources/dart2/serialization/native/native_class.mustache

```diff
@@ -141,7 +141,7 @@ class {{{classname}}} {
         {{{name}}}: json[r'{{{baseName}}}'] is List
           ? (json[r'{{{baseName}}}'] as List).map((e) =>
               {{#items.complexType}}
-              {{items.complexType}}.listFromJson(json[r'{{{baseName}}}']){{#required}}{{^isNullable}}!{{/isNullable}}{{/required}}{{^required}}{{#defaultValue}} ?? {{{.}}}{{/defaultValue}}{{/required}}
+              {{items.complexType}}.listFromJson(json[r'{{{baseName}}}']){{#required}}{{^isNullable}}!{{/isNullable}}{{#uniqueItems}}.toSet(){{/uniqueItems}}{{/required}}{{^required}}{{#defaultValue}} ?? {{{.}}}{{/defaultValue}}{{/required}}
```

Comment:
does it have negative performance implication that first time object will be a list and then a set? can it be a set the first time?

Reply:
I don't believe so. It's only called once and only if the user specifically asked to have a Set, and I believe falls under "normal use" when the user wants that functionality?

Reply:
>It's only called once and only if the user specifically asked to have a Set

that's one per conversion, not just once. so each time it's converting to a Set, we get a in-memory List that is then converted to a Set. Why not just create a Set in the first method call? It would eliminate unnecessary List creation and reduce GC times.

Reply:
> Why not just create a Set in the first method call?

the thing is, the library is using `jsonDecode`, which will create a List anyway since json has no concept of unique arrays